src/variableDiagnosticsProvider.ts

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+import { workspace, languages, Diagnostic, DiagnosticSeverity, DiagnosticCollection, TextDocument, Range, Position, Disposable } from 'vscode';
+
+import { event } from "./events/requestVariableEvent";
+import { VariableProcessor } from "./variableProcessor";
+
+export class VariableDiagnosticsProvider {
+
+    private diagnosticCollection: DiagnosticCollection;
+
+    public activate(subscriptions: Disposable[]) {
+        this.diagnosticCollection = languages.createDiagnosticCollection();
+
+        workspace.onDidOpenTextDocument(this.checkVariables, this, subscriptions);
+        workspace.onDidCloseTextDocument((textDocument) => {
+            this.diagnosticCollection.delete(textDocument.uri);
+        }, null, subscriptions);
+        workspace.onDidSaveTextDocument(this.checkVariables, this, subscriptions);
+
+        // Check all open documents
+        this.checkVariablesInAllTextDocuments();
+
+        event(() => {
+            this.checkVariablesInAllTextDocuments();
+        });
+    }
+
+    public dispose(): void {
+        this.diagnosticCollection.clear();
+        this.diagnosticCollection.dispose();
+    }
+
+    public async checkVariablesInAllTextDocuments() {
+        workspace.textDocuments.forEach(this.checkVariables, this);
+    }
+
+    private async checkVariables(document: TextDocument) {
+        if (document.languageId !== 'http') {
+            return;
+        }
+
+        let diagnostics: Diagnostic[] = [];
+
+        let vars = this.findVariables(document);
+
+        let varNames = vars.map((v) => v.variableName.split(".")[0].split("[")[0]);
```

Comment:
Why do we need an extra split for `[`?

Reply:
I think this is some old logic I forgot to remove, we don't need the split for either `.` or `[`. Fixing.